applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp

```diff
@@ -399,65 +399,125 @@ namespace Kratos {
     } //CreateContactElements
 
     void ContinuumExplicitSolverStrategy::SetCoordinationNumber(ModelPart& r_model_part) {
+
         ProcessInfo& r_process_info = r_model_part.GetProcessInfo();
 
-        const double in_coordination_number = r_process_info[COORDINATION_NUMBER];
+        const double desired_coordination_number = r_process_info[COORDINATION_NUMBER];
         double standard_dev = 0;
-        double out_coordination_number = ComputeCoordinationNumber(standard_dev);
+        double current_coordination_number = ComputeCoordinationNumber(standard_dev);
         int iteration = 0;
         int maxiteration = 100;
         double& added_search_distance = r_process_info[SEARCH_RADIUS_INCREMENT];
+        const bool local_coordination_option = r_process_info[LOCAL_COORDINATION_NUMBER_OPTION];
+        const bool global_coordination_option = r_process_info[GLOBAL_COORDINATION_NUMBER_OPTION];
         added_search_distance = 0.0;
         double amplification = 1.0;
 
         if(r_model_part.GetCommunicator().MyPID() == 0) {
-            KRATOS_WARNING("DEM") << "Setting up Coordination Number by increasing or decreasing the search radius... " << std::endl;
-        }
+            KRATOS_INFO("DEM") << "Setting up Coordination Number (input = "<<desired_coordination_number<<") by increasing or decreasing the search radius. ";
+            if(local_coordination_option) KRATOS_INFO("") << "Local extension activated. ";
+            if(global_coordination_option) KRATOS_INFO("") << "Global extension activated. ";
+            KRATOS_INFO("DEM") << std::endl;
+        }
+
+        if (desired_coordination_number <= 0.0) {
+            KRATOS_THROW_ERROR(std::runtime_error, "The specified Coordination Number is less or equal to zero, N.C. = ", desired_coordination_number)
+        }
+
+        //STAGE 1, Local Coordination Number
+        double tolerance = 0.5;
+        double max_factor_between_iterations = 1.02;
+        double relative_error = 1000;
```

Comment:
I guess those can be `const`

Reply:
They can't, they are overwritten some lines below.